pkg/account/handler.go

```diff
@@ -0,0 +1,202 @@
+package account
+
+import (
+	"container/list"
+	"crypto/rand"
+	"net/http"
+	"regexp"
+
+	"github.com/labstack/echo/v4"
+)
+
+var account_list = list.New()
+
+// type AcountType string
+
+type AccountHandler struct {
+}
+
+// @Summary create and account based on email(as id), type, name and password
+// @Description will check primarykey other, then add to account_list if possible
+// @Tags Account
+// @Produce json
+// @Param email formData string true "user e-mail"
+// @Param type formData string true "user type"
+// @Param name formData string true "user name"
+// @Param paswd formData string true "user password"
+// @Success 200 {string} string "{"msg": "Successfully created"}"
+// @Failure 400 {string} string "{"error": "Invali E-mail Address"}"
+// @Router /case/PastHistory/{id} [PATCH]
+func (h *AccountHandler) CreateAccount(c echo.Context) error {
+	Email := c.QueryParam("Email")
+	Type := AcountType(c.QueryParam("Type"))
+	Name := c.QueryParam("Name")
+	Passwd := c.QueryParam("Passwd")
+
+	if ok, _ := regexp.MatchString(`^\w+@\w+[.\w+]+$`, Email); !ok {
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": "Invali E-mail Address"})
+	}
+
+	if Type != ACCOUNT_TYPE_PATIENT && Type != ACCOUNT_TYPE_DOCTOR && Type != ACCOUNT_TYPE_ADMIN {
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": "Invali Account Type"})
+	}
+
+	if len(Passwd) < ACCOUNT_PASSWD_LEN {
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": "Invali Password Length"})
+	}
+
+	// Check uniqueness
+	for itor := account_list.Front(); itor != nil; itor = itor.Next() {
+		if itor.Value.(Account).Email == Email {
+			return c.JSON(http.StatusBadRequest, map[string]string{
+				"error": "E-Mail occupied"})
+		}
+	}
+
+	account_list.PushBack(Account{Email: Email, Type: Type, Name: Name, Passwd: Passwd})
+
+	return c.JSON(http.StatusOK, map[string]string{
+		"msg": "Successfully created"})
+
+}
+
+/**
+ * @todo cookie not implemented
```

Comment:
I suggest using jwt, but it's ok if you insist on cookie. BTW, your group is resonsible for writing a middleware for auth.